src/mds/MDSCacheObject.h

```diff
@@ -119,6 +120,7 @@ class MDSCacheObject {
   void state_set(unsigned mask) { state |= mask; }
   void state_reset(unsigned s) { state = s; }
 
+  bool is_dir() const { return state_test(STATE_DIR); }
```

Comment:
better would be a `virtual bool is_dir() const` that is overriden here (and defined in `MDSCacheObject`.

Reply:
Sure.